### PR TITLE
Make publish_version take in an optional alias file

### PR DESCRIPTION
**Pull request by krzyzacy**
I'm a bash noob, and not sure how to test things properly, but this is the idea for publishing to additional alias txt for ci builds.

/assign @david-mcmahon @ixdy 

xref https://github.com/kubernetes/test-infra/pull/3490

### CODE REVIEW

File: push-build.sh [hunk 1]
```diff
@@ -42,7 +42,8 @@ PROG=${0##*/}
 #+ OPTIONS
 #+     [--nomock]                - Enables a real push (--ci only)
 #+     [--federation]            - Enable FEDERATION push
-#+     [--ci]                    - Used when called from Jekins (for ci runs)
+#+     [--ci]                    - Used when called from Jenkins (for ci runs)
+#+     [--alias=]                - Used when need to upload to version alias txt (--ci only)
```
Comment (david-mcmahon) [inline push-build.sh]: I don't think `alias` is descriptive enough here.  How about `--extra-publish-file`.  Also the man page should describe the format of the arg which is a relative path that is appended to a GCS path.  That should be noted so the user understands how the arg is constructed and used.

File: push-build.sh [hunk 2]
```diff
@@ -146,6 +147,7 @@ GCS_DEST="devel"
 ((FLAGS_ci)) && GCS_DEST="ci"
 GCS_DEST=${FLAGS_release_type:-$GCS_DEST}
 GCS_DEST+="$FLAGS_gcs_suffix"
+GCS_ALIAS=${FLAGS_alias:-}
```
Comment (david-mcmahon) [inline push-build.sh]: GCS_EXTRA_PUBLISH_FILE

File: lib/releaselib.sh [hunk 3]
```diff
@@ -594,6 +594,7 @@ release::gcs::publish_version () {
   local version=$2
   local build_output=$3
   local bucket=$4
+  local build_alias=$5
```
Comment (david-mcmahon) [inline lib/releaselib.sh]: local extra_publish_file

File: lib/releaselib.sh [hunk 4]
```diff
@@ -620,6 +621,9 @@ release::gcs::publish_version () {
                  $type-$version_major
                  $type-$version_major.$version_minor
                 )
+  if [[ -n "${build_alias:-}" ]]; then
```
Comment (david-mcmahon) [inline lib/releaselib.sh]: you can just add $local_publish_file to the array above.  It'll only add to the array if it's set.

**Comment by krzyzacy** [comment]
@david-mcmahon renamed the flag, PTAL

**Comment by krzyzacy** [comment]
thanks :-) what's the merge process in this repo?

**Comment by david-mcmahon** [comment]
you can merge at will.

**Comment by krzyzacy** [comment]
I don't have merge here :-( can you help?

**Comment by krzyzacy** [comment]
thanks!

**Pull request merged**